api/ruxos_posix_api/src/imp/signal.rs

```diff
@@ -86,8 +86,24 @@ pub unsafe fn sys_sigaltstack(
     syscall_body!(sys_sigaltstack, Ok(0))
 }
 
-/// TODO: send a signal to a process
-pub unsafe fn sys_kill(pid: pid_t, sig: c_int) -> c_int {
+/// send a signal to a process
+pub fn sys_kill(pid: pid_t, sig: c_int) -> c_int {
     debug!("sys_kill <= pid {} sig {}", pid, sig);
-    syscall_body!(sys_kill, Ok(0))
+    syscall_body!(sys_kill, {
+        match Signal::signal(sig as _, true) {
+            None => Err(LinuxError::EINVAL),
+            Some(_) => Ok(0),
+        }
+    })
+}
+
+/// send a signal to a thread
+pub fn sys_tkill(tid: pid_t, sig: c_int) -> c_int {
```

Comment:
How does this `sys_tkill` send a signal to a thread ranther than a process? It seems that no changes have been made with `sys_kill`.